src/app/components/goal-management/goal-management.component.ts

```diff
@@ -8,8 +11,16 @@ import {FormControl} from '@angular/forms';
 })
 export class GoalManagementComponent implements OnInit {
   date = new FormControl(new Date());
+  goalSet: Goal[];
+  userName: string;
 
-  constructor() {}
+  constructor(
+      private goalManagementService: FakeGoalManagementService,
+      private route: ActivatedRoute) {}
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.userName = this.route.params['username'];
```

Comment:
Make this a static const. You can also define it in the routes definition file and reference it from here.

Reply:
No idea how to do it.